src/packages/ratmama/ratmama_parser.py

```diff
@@ -167,39 +172,48 @@
         sep = '-'
 
     if not sep:
-        ctx.bot.board.append(Rescue(irc_nickname=ctx.user.nickname, client=ctx.user.nickname))
+        ctx.bot.board.append(Rescue(irc_nickname=ctx.user.nickname.split("[")[0],
+                                    client=ctx.user.nickname))
         index = ctx.bot.board.find_by_name(ctx.user.nickname)
         await ctx.reply(f"Case #{index} created for {ctx.user.nickname}, please set details")
         return
     system: str = None
-    code_red: bool = False
+    cr: bool = False
     platform: Platforms = None
     for part in message.split(sep):
         part = part.strip()
         if part.casefold() in ("pc",):
             platform = Platforms["PC"]
+            message = re.sub(part, "", message, flags=re.IGNORECASE)
 
         elif part.casefold() in ("ps", "ps4", "playstation", "playstation4", "playstation 4"):
             platform = Platforms["PS"]
+            message = re.sub(part, "", message, flags=re.IGNORECASE)
 
         elif part.casefold() in ("xb", "xb1", "xbox", "xboxone", "xbox one"):
             platform = Platforms["XB"]
+            message = re.sub(part, "", message, flags=re.IGNORECASE)
 
         elif "o2" in part.casefold():
-            code_red = "o2 ok" not in part.casefold()
+            cr = "o2 ok" not in part.casefold()
+            message = re.sub(part, "", message, flags=re.IGNORECASE)
 
-        else:
-            system = part
+    system = message.strip(f"{sep} ")
 
     rescue = Rescue(
-        client=ctx.user.nickname,
+        client=ctx.user.nickname.split("[")[0],
```

Comment:
[suggestion] this expression, `.split('[')[0]`, while simple, is repeated at least twice within this module. Further, i can forsee usages of this expression in other modules/

Therefore, it may be wise to move this routine into our utilities package
```py
def strip_tags(nickname:str) -> str
    return nickname.strip('[')[0]
```

Reply:
looking into ratlib, this function actually exists already

Reply:
I could have sworn it had existed, but evidently i didn't look hard enough when i searched for it prior to rendering this suggestion 😅 

```py
def strip_name(nickname: str) -> str:
    """
    This function accepts one input `nickname` and returns the input string
    minus any tags.

    An IRC tag is anything starting with "`[`". Further, anything following a
    `[` is truncated.

    Args:
        nickname (str): raw nickname to strip tags

    Returns:
        str: nickname stripped of tags
    """
    split_string = nickname.split("[")
    return split_string[0]

``` so it does